service/src/main/java/com/codeforcommunity/processor/SiteProcessorImpl.java

```diff
@@ -93,6 +108,261 @@ public void recordStewardship(
   }
 
   @Override
+  public void updateSite(JWTData userData, int siteId, UpdateSiteRequest updateSiteRequest) {
+    checkSiteExists(siteId);
+
+    SiteEntriesRecord lastEntry =
+        db.selectFrom(SITE_ENTRIES)
+            .where(SITE_ENTRIES.SITE_ID.eq(siteId))
+            .orderBy(SITE_ENTRIES.UPDATED_AT.desc())
+            .limit(1)
+            .fetchOne();
+
+    SiteEntriesRecord record = db.newRecord(SITE_ENTRIES);
+
+    record.setUserId(userData.getUserId());
+    record.setSiteId(siteId);
+    record.setUpdatedAt(new Timestamp(System.currentTimeMillis()));
+    if (updateSiteRequest.isTree_present() != null) {
```

Comment:
I think the idea is that users will be given a filled in form on the front end with the latest update, and then they can change anything they want. With the implementation you've used it would be impossible to set fields to null if that is the accurate state based on their assessment.